File: Python/347_Top_K_Frequent_Elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d = {}
        for i in nums:
            if i not in d:
                d[i] = 1
            else:
                d[i] += 1
        l = []
        for i in range(len(d)+1):
            l.append([])
        for i in d:
            l[d[i] - 1].append(i)
        a = []
        for i in range(len(l) - 1, -1, -1):
            a += l[i]
        return a[0:k]

File: C++/347_Top_K_Frequent_Elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        priority_queue<pair<int, int>> pq;
        vector<int> ans;
        for(int &x : nums){
            freq[x]++;
        }
        for(auto &y : freq){
            pq.push({y.second, y.first});
        }
        for(int i = 0; i < k; i++){
            pair<int, int> top = pq.top();
            pq.pop();
            ans.push_back(top.second);
        }
        return ans;
    }
};